### PR TITLE
docs: outline go support policy

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** <Describe what has changed.>

**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: https://github.com/open-telemetry/opentelemetry-operator/issues/2757

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -714,50 +714,6 @@ spec:
           exporters: [debug]
 ```
```
Comment (swiatekm-sumo) [inline README.md:716]: We should link to support.md here.

File: support.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Support
+
+## Go
+
+When productised as a go libary or custom distribution the OpenTelemetry Operator project attempts to follow the supported go versions as [defined by the Go team](https://go.dev/doc/devel/release#policy).
```
Comment (swiatekm-sumo) [inline support.md:5]: How about:

```suggestion
When used as a Go libary or custom distribution, the OpenTelemetry Operator project supports at minimum all the Go versions [supported by the Go team](https://go.dev/doc/devel/release#policy) at any given time.
```

"productised" is a strange word to use in this context imo.

File: support.md
Comment (pavolloffay) [inline support.md:1]: Can we rename this to `SUPPORT.md` to align with other files?

File: support.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+We strive to be compatible with the widest range of Kubernetes versions as possible, but some changes to Kubernetes itself require us to break compatibility with older Kubernetes versions, be it because of code incompatibilities, or in the name of maintainability. Every released operator will support a specific range of Kubernetes versions, to be determined at the latest during the release.
+
+We use `cert-manager` for some features of this operator and the third column shows the versions of the `cert-manager` that are known to work with this operator's versions.
+
+The OpenTelemetry Operator _might_ work on versions outside of the given range, but when opening new issues, please make sure to test your scenario on a supported version.
+
+| OpenTelemetry Operator | Kubernetes     | Cert-Manager |
+| ---------------------- | -------------- | ------------ |
+| v0.97.0                | v1.23 to v1.29 | v1           |
```
Comment (pavolloffay) [inline support.md:36]: This needs to be updated

File: support.md
Comment (swiatekm-sumo) [reply support.md:1]: It's aligned with `api.md`. I don't think there's a way to rename that?

**Comment by frzifus** [comment]
Ouch, let me update this...

**Comment by jaronoff97** [comment]
@frzifus any chance you can dothe updates, would love to merge this.

**Comment by frzifus** [comment]
Yes, its on my list. I try to provide an update next week.

**Comment by frzifus** [comment]
cc @jaronoff97 @swiatekm

**Review by swiatekm — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Comment by swiatekm** [comment]
@frzifus can you rebase this on main after #3393, and then we could finally merge?

**Comment by frzifus** [comment]
oh yes, please.

**Comment by frzifus** [comment]
The PR is now much shorter. The majority was already moved into `docs/compatibility.md`.

File: README.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ The operator manages:
 
 ## Documentation
 
+- [Compatibility & Support docs](./docs/compatibility.md)
```
Comment (swiatekm) [inline README.md:14]: If we're adding this, I think we should also remove https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#compatibility

**Review by swiatekm — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]

**Pull request merged**